apiFunctions.php

```diff
@@ -509,6 +509,38 @@ function expand_by_jstor($template) {
   return TRUE;
 }
 
+function expand_researchgate($templates) {
+  foreach ($templates as $template) expand_by_researchgate($template);
+}
+
+function expand_by_researchgate($template, $url_sent = NULL) {
+  if ($template->incomplete() === FALSE) return FALSE;
+  if (is_null($url_sent)) $url_sent = $template->get('url');
+  if (preg_match("~^https?://(?:www.|)researchgate.net/[^\s]*publication/([0-9]+)~i", $url_sent, $match)) {
+    $url = 'https://www.researchgate.net/publicliterature.PublicationHeaderDownloadCitation.downloadCitation.html?publicationUid=' . $match[1] . '&fileType=RIS&citationAndAbstract=false'; // Convert researchgate URL to give RIS information
+  } else {
+    return FALSE;
+  }
+  $cookie_opts = array(
+    'http'=>array(
+      'method'=>"GET",
+      'follow_location' => true,
+      'header'=>"Accept-language: en\r\n" .
+                "Cookie: cookieconsent_dismissed=true\r\n" . 
+                "User-Agent: Mozilla/5.0 (Citation Bot)\r\n" .
```

Comment:
Can we use a globally defined User-Agent?  We ought to have this in a constant somewhere...